### PR TITLE
Bug/2039 let submit button inactive if there are not filled required fields within create forms

### PR DESCRIPTION
**Pull request by lazzy-panda**
Added correct validation for Create New Entities forms with some refactoring

**How Has This Been Tested?** (put an "x" (case-sensitive!) next to an item)

- [ ] No need to
- [x] Manually (please, describe, if necessary)
- [x] Unit checks
- [ ] Integration checks
- [x] Covered by existing automation

**Checklist** (put an "x" (case-sensitive!) next to all the items, otherwise the build will fail)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [x] My changes generate no new warnings (e.g. Sonar is happy)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged

Check out [Contributing](https://github.com/provectus/kafka-ui/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/provectus/kafka-ui/blob/master/CODE-OF-CONDUCT.md)

**A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: kafka-ui-react-app/src/components/Topics/Topic/SendMessage/SendMessage.tsx [hunk 1]
```diff
@@ -92,22 +86,19 @@ const SendMessage: React.FC = () => {
...
-    partition: number;
-  }) => {
+  console.log(keyDefaultValue);
+
+  const [key, setKey] = useState(keyDefaultValue);
+  const [content, setContent] = useState(contentDefaultValue);
+  const [headers, setHeaders] = useState({});
+  const [partition, setPartition] = useState(0);
+
+  const isValid = key && content && headers;
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/Topic/SendMessage/SendMessage.tsx:96]: Why we trying to reimplement `isValid` method of `react-hook-form`?

File: kafka-ui-react-app/src/components/Topics/Topic/SendMessage/SendMessage.tsx [hunk 2]
```diff
@@ -92,22 +86,19 @@ const SendMessage: React.FC = () => {
...
     if (messageSchema) {
-      const { partition, key, content } = data;
       const errors = validateMessage(key, content, messageSchema);
-      if (data.headers) {
-        try {
-          JSON.parse(data.headers);
-        } catch (error) {
-          errors.push('Wrong header format');
-        }
-      }
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/Topic/SendMessage/SendMessage.tsx:110]: Are you sure we don't need that anymore?

File: kafka-ui-react-app/src/components/Topics/Topic/SendMessage/validateMessage.ts [hunk 3]
```diff
@@ -3,7 +3,7 @@ import Ajv from 'ajv/dist/2020';
 import { upperFirst } from 'lodash';
 
 const validateBySchema = (
-  value: string,
+  value: string | undefined,
```
Comment (Mgrdich) [inline kafka-ui-react-app/src/components/Topics/Topic/SendMessage/validateMessage.ts:6]: can't we just write 
`value?: string ,`

**Review by Mgrdich — COMMENTED** [review]
and remove package-json lock

File: kafka-ui-react-app/src/components/Topics/Topic/SendMessage/validateMessage.ts [hunk 4]
```diff
@@ -46,8 +46,8 @@ const validateBySchema = (
 };
 
 const validateMessage = (
-  key: string,
-  content: string,
+  key: undefined | string,
+  content: undefined | string,
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/Topic/SendMessage/validateMessage.ts:50]:
```suggestion
  content?: string,
```

File: kafka-ui-react-app/src/components/Topics/Topic/SendMessage/validateMessage.ts (hunk 3 above)
Comment (lazzy-panda) [reply kafka-ui-react-app/src/components/Topics/Topic/SendMessage/validateMessage.ts:6]: fixed

File: kafka-ui-react-app/src/components/Topics/Topic/SendMessage/SendMessage.tsx (hunk 1 above)
Comment (lazzy-panda) [reply kafka-ui-react-app/src/components/Topics/Topic/SendMessage/SendMessage.tsx]: Fixed

File: kafka-ui-react-app/src/components/Topics/Topic/SendMessage/SendMessage.tsx [hunk 5]
```diff
@@ -92,22 +86,19 @@ const SendMessage: React.FC = () => {
     });
   }, [keyDefaultValue, contentDefaultValue, reset]);
 
-  const onSubmit = async (data: {
-    key: string;
-    content: string;
-    headers: string;
-    partition: number;
-  }) => {
+  console.log(keyDefaultValue);
```
[the remark this answers is not in the document]
Comment (lazzy-panda) [reply kafka-ui-react-app/src/components/Topics/Topic/SendMessage/SendMessage.tsx]: ok

File: kafka-ui-react-app/src/components/Topics/New/__test__/New.spec.tsx [hunk 6]
```diff
@@ -105,8 +105,13 @@ describe('New', () => {
     });
 
     await waitFor(() => {
+      const btn = screen.getByText(/submit/i);
       userEvent.type(screen.getByPlaceholderText('Topic Name'), topicName);
-      userEvent.click(screen.getByText(/submit/i));
+      userEvent.type(screen.getByPlaceholderText('Number of partitions'), '1');
+      btn.removeAttribute('disabled');
```
[the remark this answers is not in the document]
Comment (lazzy-panda) [reply kafka-ui-react-app/src/components/Topics/New/__test__/New.spec.tsx]: In this cases I can't make this button enabled by userEvent actions, I think I need some help to handle this

File: kafka-ui-react-app/src/components/Topics/New/__test__/New.spec.tsx [hunk 7]
```diff
@@ -105,8 +105,13 @@ describe('New', () => {
...
+
+      expect(btn).toBeEnabled();
+      userEvent.click(btn);
```
[the remark this answers is not in the document]
Comment (lazzy-panda) [reply kafka-ui-react-app/src/components/Topics/New/__test__/New.spec.tsx]: fixed

File: kafka-ui-react-app/src/components/Topics/Topic/Edit/__test__/Edit.spec.tsx [hunk 8]
```diff
@@ -141,10 +144,7 @@ describe('Edit Component', () => {
       const btn = screen.getAllByText(/submit/i)[0];
 
       await act(() => {
-        userEvent.type(
-          screen.getByPlaceholderText('Min In Sync Replicas'),
-          '1'
-        );
+        btn.removeAttribute('disabled');
```
[the remark this answers is not in the document]
Comment (lazzy-panda) [reply kafka-ui-react-app/src/components/Topics/Topic/Edit/__test__/Edit.spec.tsx]: Fixed

File: kafka-ui-react-app/src/components/Topics/New/__test__/New.spec.tsx [hunk 9]
```diff
@@ -105,8 +105,13 @@ describe('New', () => {
     });
 
     await waitFor(() => {
+      const btn = screen.getByText(/submit/i);
       userEvent.type(screen.getByPlaceholderText('Topic Name'), topicName);
-      userEvent.click(screen.getByText(/submit/i));
+      userEvent.type(screen.getByPlaceholderText('Number of partitions'), '1');
+      btn.removeAttribute('disabled');
+
+      expect(btn).toBeEnabled();
```
[the remark this answers is not in the document]
Comment (lazzy-panda) [reply kafka-ui-react-app/src/components/Topics/New/__test__/New.spec.tsx]: fixed

File: kafka-ui-react-app/src/components/Topics/New/__test__/New.spec.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (lazzy-panda) [reply kafka-ui-react-app/src/components/Topics/New/__test__/New.spec.tsx]: Fixed

File: kafka-ui-react-app/.gitignore [hunk 10]
```diff
@@ -18,6 +18,8 @@ build
 .env.development.local
 .env.test.local
 .env.production.local
+.gitignore
+package-lock.json
```
Comment (workshur) [inline kafka-ui-react-app/.gitignore:22]: ?

Comment (lazzy-panda) [reply kafka-ui-react-app/.gitignore]: removed

File: kafka-ui-react-app/src/components/Topics/Topic/SendMessage/SendMessage.tsx [hunk 11]
```diff
@@ -167,83 +155,86 @@ const SendMessage: React.FC = () => {
...
-                  onChange={onChange}
-                  minWidth="100%"
-                  options={selectPartitionOptions}
-                  value={selectPartitionOptions[0].value}
-                />
-              )}
-            />
-          </S.Column>
-        </S.Columns>
+        <div className="columns">
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/Topic/SendMessage/SendMessage.tsx:158]: We use `styled-components`. There should be no classes

**Review by workshur — CHANGES_REQUESTED** [review]

**Comment by Haarolean** [comment]
Done within #2221

**Pull request closed without merging**